### PR TITLE
ommongodb: rewrite documentation

### PR DESCRIPTION
**Pull request by PascalWithopf**
Module documentation was rewritten in the new style.

### CODE REVIEW

File: source/configuration/modules/ommongodb.rst [hunk 1]
```diff
@@ -1,56 +1,176 @@
...
+   :header: "type", "default", "mandatory", "|FmtObsoleteName| directive"
+   :widths: auto
+   :class: parameter-table
+
+   "word", "none", "no", "none"
+
+Absolute path to the trusted X509 CA certificate that signed the mongoDB server certificate. This is optional.
+
+
+(Server) deprecated, use "UriStr" instead
```
Comment (PascalWithopf) [inline source/configuration/modules/ommongodb.rst]: How should I handle deprecated configuration parameter in the new style?

**Comment by deoren** [comment]
> @PascalWithopf: How should I handle deprecated configuration parameter in the new style?

Good question, and sorry for the radio silence (time is scarce for me right now).

Perhaps handle it like with the `imfile` doc?

http://www.rsyslog.com/doc/v8-stable/configuration/modules/imfile.html#deprecated-parameters

I recall a discussion of having a central list vs keeping a list on each module page, but since we've got at least one other module doc with the deprecated parameters on the same page I vote that we go with that approach for now.

**Comment by PascalWithopf** [comment]
@deoren I will change it like imfile and if we want to create a central list we can still do this later.

**Comment by deoren** [comment]
> @PascalWithopf: @deoren I will change it like imfile and if we want to create a central list we can still do this later.

Acknowledged.

Are all four of these parameters deprecated?

- Server
- ServerPort (shown in 55942da351e5e8bcb945b2ed937c6ba443f9287d as `ServerPorted`)
- UID
- PWD

I ask because the "Deprecated Parameters" section is just above the "Action Parameters" section where those four parameters are listed in 55942da351e5e8bcb945b2ed937c6ba443f9287d.

**Comment by PascalWithopf** [comment]
@deoren Yes they are all deprecated. They are replaced by the parameter `UriStr`.
`ServerPorted` is a typo.
And I created the Deprecated Parameters section at the end of the file separate of the other parameters.

**Comment by deoren** [comment]
> @PascalWithopf: @deoren Yes they are all deprecated. They are replaced by the parameter UriStr.
> ServerPorted is a typo.
> And I created the Deprecated Parameters section at the end of the file separate of the other parameters.

I see now. When I glanced quickly it appeared that the Deprecated Parameters section was sandwiched between two other sections, but I see now that the 'Action Parameters' section there is a sub-section. Do we need that subsection? I wonder if it's perhaps subdividing what is already a short list of content.

**Comment by deoren** [comment]
@PascalWithopf I saw your agreement on a different PR re pushing back to your branch, so I took a stab at introducing some changes here. Please let me know what you think.

**Comment by PascalWithopf** [comment]
@deoren I think you are right, there is no reason to divide the Deprecated Parameters section. Thanks for the help.

**Comment by deoren** [comment]
> @PascalWithopf: @deoren I think you are right, there is no reason to divide the Deprecated Parameters section. Thanks for the help.

You are welcome. Your remarks on a different PR re the section names for the examples reminded me to take a closer look here. I adjusted the names for those sections in an effort to match the others.

Any concerns?

**Comment by PascalWithopf** [comment]
Same as here https://github.com/rsyslog/rsyslog-doc/pull/603#issuecomment-373403028

**Comment by deoren** [comment]
> > @PascalWithopf: @deoren I think you are right, there is no reason to divide the Deprecated Parameters section. Thanks for the help.
> 
> You are welcome. Your remarks on a different PR re the section names for the examples reminded me to take a closer look here. I adjusted the names for those sections in an effort to match the others.
> 
> Any concerns?
>
> @PascalWithopf: Same as here https://github.com/rsyslog/rsyslog-doc/pull/603#issuecomment-373403028

Do you wish to keep the generic names instead of named examples? I prefer the latter because if I'm looking at the Table of Contents in the side bar I can jump right to the example I'm looking for. I can see how it may be useful to refer to a specific generically named example from within the docs (perhaps from a parameter description), but that can be worked around by using a label reference.

**Comment by PascalWithopf** [comment]
@deoren I have changed the example headings as we discussed in https://github.com/rsyslog/rsyslog-doc/pull/603

**Comment by deoren** [comment]
> @PascalWithopf: @deoren I have changed the example headings as we discussed in #603

ACK.

**Pull request merged**